nixos/modules/system/activation/switch-to-configuration.py

```diff
@@ -0,0 +1,604 @@
+#! @python@
+
+import os.path
+import subprocess
+from syslog import openlog, syslog, LOG_NOTICE, LOG_ERR
+import sys
+
+out = "@out@"
+
+# To be robust against interruption, record what units need to be started etc.
+startListFile = "/run/systemd/start-list"
+restartListFile = "/run/systemd/restart-list"
+reloadListFile = "/run/systemd/reload-list"
+
+def die_with_usage():
```

Comment:
Would you mind to type annotations, then changing this file becomes easier for type checking with mypy.
I can also add them if you want.

Reply:
I was very tempted to add type annotations when I started but wasn't quite sure whether we would want to retain Python 2 support so (mostly) refrained. I would be happy to add them.

Reply:
No, we definitely don't want more legacy python2 code.